tests/testthat/test-knitr_formats.R

```diff
@@ -12,9 +12,9 @@ test_that("it handles dir", {
     parse_settings = FALSE
   )
   has_lints <- length(lints) > 0
-  testthat::expect(has_lints, "There should be lints")
+  expect_true(has_lints, info="There should be lints")
 
-  testthat::expect_equivalent(length(unique(names(lints))), 6, info="For every file there should be at least 1 lint")
+  expect_identical(length(unique(names(lints))), 6L, info="For every file there should be at least 1 lint")
```

Comment:
nit: There should be spaces around `=`. Also, wondering whether it's worth replacing `6L` by `length(list.files(...))` to make it more robust against the unlikely case that {knitr} gets a new format, we add it and forget about this test while it passes because linting the new file returns no lints...

Reply:
I think tests overall could go for some house-keeping. I'll leave it out of scope for this PR; filing a follow-up.

Reply:
https://github.com/r-lib/lintr/issues/918

Reply:
Thanks for filing the follow-up. Can we delint those lines anyway?

Reply:
will do